bin/main.rb

```diff
@@ -29,52 +24,49 @@
 system 'clear'
 system 'cls'
 
-box = TTY::Box.frame '+---+---+---+',
-                     '| 1 | 2 | 3 |',
-                     '+---+---+---+',
-                     '| 4 | 5 | 6 |',
-                     '+---+---+---+',
-                     '| 7 | 8 | 9 |',
-                     '+---+---+---+',
-                     padding: 1, align: :center
+board = Board.new
 
 def tie
-  puts "IT'S A TIE!\n"
-  puts 'Game Over'
+  puts TTY::Box.frame "IT'S A TIE!\nGame Over", padding: 2, align: :center
 end
 
-def win(player)
-  puts "#{player} you WIN the game!"
+def win(player, board)
+  puts board.draw_the_board
+  puts TTY::Box.frame "#{player.player[:name]} you WIN the game!\n", padding: 2
 end
 
 i = 0
 while i < 9
-
-  puts box
-
+  puts board.draw_the_board
   player = players[i % 2]
-  puts "It's #{player}'s turn!\n"
+  puts "It's #{player.player[:name]}'s turn!\n"
 
   puts 'Please select an available cell from the board.'
   received_number = ''
-  until (1..9).include?(received_number)
+  until (1..9).include?(received_number) && !board.whole_moves.include?(received_number)
     received_number = gets.chomp.to_i
     puts 'Invalid move. Please enter a number from 1-9.' unless (1..9).include?(received_number)
+    if board.whole_moves.include?(received_number)
+      puts 'Invalid move. This number has been already chose, choose new number.'
+    end
   end
-
+  player.add_moves(received_number)
+  board.add_moves(received_number)
+  board.grid[received_number - 1] = player.player[:mark]
+  puts "this the value #{board.grid[received_number - 1]}"
   system 'clear'
   system 'cls'
 
-  if i == rand(9)
-    puts box
-    win(player)
+  i += 1
+
+  if board.check_winner(player.player_moves)
+    win(player, board)
     break
   end
 
-  i += 1
 end
 
 if i == 9
-  puts box
+  puts board.draw_the_board
   tie
 end
```

Comment:
This condition should check if the game is a win first before deciding it as a TIE because players can make a winning move even at the last (i ==9) move.

Reply:
Thanks for your helpful feedback. We have worked on it.